### PR TITLE
added dirty management in download cache

### LINKED ISSUE #8578: [bug] conan tries to extract broken downloads
When `CTRL + C` while conan downloads the prebuilt archives of a package, the subsequent execution of `conan install` tries to extract from those half-downloaded archives from the `download_cache`. This does **ONLY** happens when you use a `download_cache`.

2 possible ideas:

- add a hash to check if the downloaded archives are not corrupt
- restart the download when the extraction fails

### Environment Details (include every applicable attribute)
  * Operating System+version: Ubuntu 18.04 WSL2
  * Compiler+version: gcc 7.4
  * Conan version: 1.33.1
  * Python version: 3.6.9

**relevant conan settings:** `download_cache = /tmp/conan_download_cache`

### Steps to reproduce (Include if Applicable)
Its easy to replicate with a "big" package, maybe something like opencv, that is available as prebuilt on the remote of your choice. If not, use your own conan repo, build and upload your package.

1. `conan install opencv/4.5.0@` and hit `CTRL + C` when conan says: `Downloading 834c114e736468c70cf2b5d0781c6c8db5787764b30d718ac40d45e51ec34c0d:  23%|##2       | 2.64M/11.7M`
1. rerun `conan install opencv/4.5.0@` and see:
```
Downloading binary packages in 8 parallel threads
opencv/4.5.0: Retrieving package 05ae15556d4c5439324a478fdaea5d0109fda60e from remote 'luminar' 
Decompressing conan_package.tgz:  61%|######    | 3.15M/5.18M [00:00<00:00, 33.1MB/s]opencv/4.5.0: ERROR: Exception while getting package: 05ae15556d4c5439324a478fdaea5d0109fda60e
opencv/4.5.0: ERROR: Exception: <class 'conans.errors.ConanException'> Error while downloading/extracting files to /home/fberchtold/.conan/data/opencv/4.5.0/_/_/package/05ae15556d4c5439324a478fdaea5d0109fda60e
Compressed file ended before the end-of-stream marker was reached
Folder removed
ERROR: Error while downloading/extracting files to /home/fberchtold/.conan/data/opencv/4.5.0/_/_/package/05ae15556d4c5439324a478fdaea5d0109fda60e
Compressed file ended before the end-of-stream marker was reached
Folder removed
```

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Fix: Add dirty management in download cache, so interrupted downloads doesn't need a manual cleaning of such download cache.
Docs: Omit

Close https://github.com/conan-io/conan/issues/8578

### CODE REVIEW

**Comment by blackliner** [comment]
Tried your changes locally, works 👍

**Review by blackliner — APPROVED** [review]

File: conans/test/integration/cache/download_cache_test.py [hunk 1]
```diff
@@ -148,7 +171,7 @@ def source(self):
         self.assertEqual("some query", client.load("myfile2.txt"))
 
         # 2 files cached, plus "locks" folder = 3
```
Comment (solvingj) [inline conans/test/integration/cache/download_cache_test.py:174]: Code changed magic number to 4, can we update comment as to why?

File: conans/test/integration/cache/download_cache_test.py [hunk 2]
```diff
@@ -126,7 +149,7 @@ def source(self):
         self.assertIn("ConanException: md5 signature failed for", client.out)
         self.assertIn("Provided signature: kk", client.out)
         self.assertIn("Computed signature: 9893532233caff98cd083a116b013c0b", client.out)
-        self.assertEqual(1, len(os.listdir(cache_folder)))  # Nothing was cached
+        self.assertEqual(2, len(os.listdir(cache_folder)))  # Nothing was cached
```
Comment (solvingj) [inline conans/test/integration/cache/download_cache_test.py:153]: Why is this magic number 2? Can we add comment like line 173?

Comment (memsharded) [reply conans/test/integration/cache/download_cache_test.py:153]: Done, please re-check

**Review by solvingj — APPROVED** [review]

**Pull request merged**